db/migrate/20141121200053_create_endpoints.rb

```diff
@@ -0,0 +1,13 @@
+class CreateEndpoints < ActiveRecord::Migration
+  def change
+    create_table :endpoints do |t|
+      t.string     :role
+      t.string     :ipaddress
+      t.string     :hostname
+      t.integer    :port
+      t.belongs_to :resource, :polymorphic => true, :type => :bigint
```

Comment:
Is there that information on EMS right now?  I was trying to limit the movement of data to just EMS right now.  Phase 2 will include the "extra" stuff that's in Provider.